### ISSUE TITLE
Parts of this page are note secure - use SSL for forum

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#2830]
To follow-up on https://github.com/neuropoly/spinalcordtoolbox/pull/2814#issuecomment-669163684, the site spinalcordtoolbox.com shows "Parts of this page are note secure". As pointed out by @Drulex, it is possibly related to the forum badge which points to a domain (spinalcordmri.org) that doesn't use SSL.

**Comment by Drulex** [comment]
How is spinalcordmri.org hosted? We can simply generate the SSL certificate and change the webserver config. I can help with that.

**Comment by jcohenadad** [comment]
> How is spinalcordmri.org hosted?

spinalcordmri.org redirects to https://spinalcordmri.github.io/, so it's a github page. Does that answer the question?

**Comment by Drulex** [comment]
This page has some info (see step 8 particularly): https://docs.github.com/en/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site.

Let's see if this is enough (I don't think the ssl cert name will match the domain, but maybe github does some magic under the hood).

**Comment by jcohenadad** [comment]
ok done-- i guess we have to wait a few hours now

**Comment by Drulex** [comment]
As I suspected:

![image](https://user-images.githubusercontent.com/7751484/89457231-0aa34e80-d733-11ea-854b-fc9d67d9ab9d.png).

There are two options:
- To use the Github generated SSL certificate we should redirect https://spinalcordmri.org to https://spinalcordmri.github.io. Currently this is not the case, I think probably a `cname` was setup since when visiting https://spinalcordmri.org the domain name stays the same instead of changing to spinalcordmri.github.io.
- Figure out if Github pages allow providing our own certificates. We would need to generate an SSL cert for `spinalcordmri.org` (matching the domain name).

I couldn't find info about the second option, I think 1 is simpler.

**Comment by jcohenadad** [comment]
does this look good?

on namecheap:
![image](https://user-images.githubusercontent.com/2482071/89461220-9a4bfb80-d739-11ea-99e4-b49bdff463e0.png)

on github:
- i removed the "custom domain"
![image](https://user-images.githubusercontent.com/2482071/89461442-fc0c6580-d739-11ea-96c1-2f67d34ef402.png)

it might take some time to take effect

**Comment by jcohenadad** [comment]
ok, i think i screwed up, and now neither spinalcordmri.org nor our SCT forum http://forum.spinalcordmri.org/c/sct are available.

so yes, i wouldn't mind some help @Drulex. I'll send you the credentials to namecheap and will make you admin of the GH site.

**Comment by Drulex** [comment]
After looking at how http://spinalcordmri.org is setup I understand why the above didn't work.

1. `spinalcordmri.org` is hosted by github pages at `spinalcordmri.github.io`. The domain `spinalcordmri.org` uses a `cname` entry to point at `spinalcordmri.github.io`. Apparently Github pages serves both SSL and non-SSL (can be accessed via `https://`)
2. The Github page `spinalcordmri.github.io` is configured with custom domain via CNAME entry making it point to `spinalcordmri.org`.
3. `forum.spinalcordmri.org` is hosted on a DigitalOcean droplet that is running the Discourse forum.

In our case we are ok with the github pages server, but whatever is running the Discourse forum is not configured for SSL.

Any SSL-enabled website pointing to the forum will have the warning described in this issue. It is the case for the docs, but also for the main page of spinalcordmri.org: https://spinalcordmri.org/.

So basically the only thing we need to do is to serve the forum via https.

Certificate I got while visiting https://spinalcordmri.org/software/ : 

![image](https://user-images.githubusercontent.com/7751484/89478263-80bcab00-d75d-11ea-9f40-dc9408b7e2ea.png)

It would also be a good idea to force only SSL connections, but as we saw earlier today that didn't work (https://github.com/neuropoly/spinalcordtoolbox/issues/2830#issuecomment-669462412) and github switched to serving with a different certificate..

**Comment by alexfoias** [comment]
@Drulex @jcohenadad I could configure the ssl certificate on the discourse forum as explained [here](https://meta.discourse.org/t/setting-up-https-support-with-lets-encrypt/40709). I just need to rerun `./discourse-setup`

**Comment by alexfoias** [comment]
@jcohenadad Did you change the configuration in namescheap for spinalcordmri.org ? I cannot see the subdomain for the forum as it should be. Info [here](https://github.com/spinalcordmri/spinalcordmri.github.io#setup-subdomain-in-namecheap)

**Comment by Drulex** [comment]
> @jcohenadad Did you change the configuration in namescheap for spinalcordmri.org ? I cannot see the subdomain for the forum as it should be. Info [here](https://github.com/spinalcordmri/spinalcordmri.github.io#setup-subdomain-in-namecheap)

It's there, maybe you need to press "show more" because there are multiple entries.

**Comment by Drulex** [comment]
> @Drulex @jcohenadad I could configure the ssl certificate on the discourse forum as explained [here](https://meta.discourse.org/t/setting-up-https-support-with-lets-encrypt/40709). I just need to rerun `./discourse-setup`

Sure if you feel comfortable doing it. At the moment the mapping of the forum domain to the correct IP isn't working for some reason. It think we should open a ticket with Namecheap as it's been like this since yesterday.

**Comment by alexfoias** [comment]
Agree, cause the configuration is correct.
![Screen Shot 2020-08-06 at 10 16 24 AM](https://user-images.githubusercontent.com/12564433/89542700-01b38b00-d7ce-11ea-852e-b576a9e8bade.png)
The http://159.89.119.65/ seems to work correctly, it might be on the DNS side.

I'll open a ticket.

**Comment by Drulex** [comment]
> I'll open a ticket.

Ok, for context the record was changed yesterday pm and reverted to the original setting while experimenting.

**Comment by alexfoias** [comment]
so they told me to change the host to forum and remove the domain and keep the same value. I changed it we'll have to wait 30 min.

**Comment by jcohenadad** [comment]
working now 🎉

**Comment by alexfoias** [comment]
I re ran the setup for discourse, now it should be secured. The only issue is the DNS that we still have to wait for.

**Comment by alexfoias** [comment]
![Screen Shot 2020-08-06 at 10 57 59 AM](https://user-images.githubusercontent.com/12564433/89547413-b69c7680-d7d3-11ea-9155-0acb64ba4e19.png)

**Comment by alexfoias** [comment]
It seems like now they create a chrome app from discourse which is cool. you can see on top at the address bar.

**Comment by jcohenadad** [comment]
yay! working now. thanks @alexfoias !

**Comment by jcohenadad** [comment]
hum, this guy is still unsecured:
![image](https://user-images.githubusercontent.com/2482071/89548302-d7190080-d7d4-11ea-8b50-f04767f78f6f.png)

**Comment by Drulex** [comment]
Looking at the requests:

![image](https://user-images.githubusercontent.com/7751484/89548899-a4233c80-d7d5-11ea-8175-5dc8945064be.png)

It's the mailing list badge img request that is unsecure: http://img.shields.io/badge/mailing%20list-development-green.svg?style=flat

**Comment by jcohenadad** [comment]
> Looking at the requests:

how do you do that?

> It's the mailing list badge img request that is unsecure

maybe we could find badge from somewhere else, e.g. herokuapp. The weird thing is that other badges are using img.shield.io, e.g. https://img.shields.io/discourse/status?label=forum&server=http%3A%2F%2Fforum.spinalcordmri.org

**Comment by alexfoias** [comment]
if you replace the badge img with this seems to be secured https://img.shields.io/badge/mailing%20list-development-green.svg?style=flat

**Comment by jcohenadad** [comment]
thanks @alexfoias ! let me try that

**Comment by Drulex** [comment]
> how do you do that?

On Firefox: Developer tools -> network monitor -> network request details.

https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/request_details

It's similar if you are using chrome

**Comment by jcohenadad** [comment]
yay! 
![image](https://user-images.githubusercontent.com/2482071/89554537-ffa4f880-d7dc-11ea-91a5-c3d95d3ed378.png)

**Comment by jcohenadad** [comment]
closing because fixed
